File: seller-web/src/types/product.ts
export interface Product {
  id: string;
  sellerId: string;
  sellerName: string;
  category: Category;
  name: string;
  nameHi?: string;
  sku: string;
  description?: string;
  unit: ProductUnit;
  minOrderQty: number;
  maxOrderQty?: number;
  imageUrls: string[];
  attributes?: Record<string, unknown>;
  status: ProductStatus;
  currentPrice?: Price;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: string;
  code: string;
  name: string;
  nameHi?: string;
  hsnCode?: string;
  description?: string;
  gstRate: number;
}

export interface Price {
  id: string;
  productId: string;
  basePrice: number;
  bulkDiscountSlabs?: BulkDiscountSlab[];
  effectiveFrom: string;
  effectiveTo?: string;
  isCurrentlyActive: boolean;
  isScheduled: boolean;
  createdAt: string;
}

export interface BulkDiscountSlab {
  minQty: number;
  maxQty?: number;
  discountPercent: number;
}

export type ProductUnit = "KG" | "PIECE" | "TRAY" | "BAG" | "BOTTLE" | "BOX";
export type ProductStatus = "ACTIVE" | "INACTIVE" | "OUT_OF_STOCK" | "DELETED";

export interface CreateProductRequest {
  categoryCode: string;
  name: string;
  nameHi?: string;
  sku: string;
  description?: string;
  unit: ProductUnit;
  minOrderQty: number;
  maxOrderQty?: number;
  imageUrls?: string[];
  attributes?: Record<string, unknown>;
  basePrice: number;
  bulkDiscountSlabs?: BulkDiscountSlab[];
}

export interface UpdateProductRequest {
  name?: string;
  nameHi?: string;
  description?: string;
  minOrderQty?: number;
  maxOrderQty?: number;
  imageUrls?: string[];
  attributes?: Record<string, unknown>;
  status?: ProductStatus;
}

export interface SetPriceRequest {
  basePrice: number;
  bulkDiscountSlabs?: BulkDiscountSlab[];
  effectiveFrom?: string;
}

File: buyer-ios/PoultryBuyer/Models/Product/Product.swift
import Foundation

struct Product: Codable, Identifiable {
    let id: String
    let sellerId: String
    let sellerName: String
    let category: Category
    let name: String
    let nameHi: String?
    let sku: String
    let description: String?
    let unit: ProductUnit
    let minOrderQty: Double
    let maxOrderQty: Double?
    let imageUrls: [String]
    let status: ProductStatus
    let currentPrice: Price?
    let createdAt: Date
    let updatedAt: Date
}

struct Category: Codable, Identifiable {
    let id: String
    let code: String
    let name: String
    let nameHi: String?
    let hsnCode: String?
    let description: String?
    let gstRate: Double
}

struct Price: Codable, Identifiable {
    let id: String
    let productId: String
    let basePrice: Double
    let bulkDiscountSlabs: [BulkDiscountSlab]?
    let effectiveFrom: Date
    let effectiveTo: Date?
    let isCurrentlyActive: Bool
    let isScheduled: Bool
    let createdAt: Date
}

struct BulkDiscountSlab: Codable {
    let minQty: Double
    let maxQty: Double?
    let discountPercent: Double
}

enum ProductUnit: String, Codable {
    case kg = "KG"
    case piece = "PIECE"
    case tray = "TRAY"
    case bag = "BAG"
    case bottle = "BOTTLE"
    case box = "BOX"
}

enum ProductStatus: String, Codable {
    case active = "ACTIVE"
    case inactive = "INACTIVE"
    case outOfStock = "OUT_OF_STOCK"
    case deleted = "DELETED"
}
